staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -785,6 +795,9 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithWarningRecorder(handler)
 	handler = genericapifilters.WithCacheControl(handler)
 	handler = genericfilters.WithHSTS(handler, c.HSTSDirectives)
+	if c.KeepListeningDuringGracefulTermination {
+		handler = genericfilters.WithRetryAfter(handler, c.terminationSignals.AfterShutdownDelayDuration.Signaled())
```

Comment:
please add a comment that this handler must be **before** the `WithWaitGroup` otherwise the server can deadlock